cmd/modelcmd/modelcommand_test.go

```diff
@@ -67,6 +77,16 @@ var modelCommandModelTests = []struct {
 	args:             []string{"--model", "explicit"},
 	expectController: "foo",
 	expectModel:      "explicit",
+}, {
+	about:            "implicit controller, explicit model UUID, short form",
+	args:             []string{"-m", "uuidfoo3"},
+	expectController: "foo",
+	expectModel:      "uuidfoo3",
+}, {
+	about:            "implicit controller, explicit model UUID, long form",
+	args:             []string{"--model", "uuidfoo3"},
+	expectController: "foo",
+	expectModel:      "uuidfoo3",
```

Comment:
these seem to test that you find the 'uuidfoo3' value, though 'uuidfoo3' is not a valid UUID, and looks like they would be going via the 'name' route.

Do we have a test that actually tracks we are able to resolve the model object from its UUID?
The code seems ok, but I don't see how that code is exercised by these tests.

I also don't see any test that uses the full UUID.

I also don't see anything about what we do if you specified "deadbeef-" or "deadbeef-abc' etc. I'm not sure what we should do there (its a valid prefix, just not a 8 character prefix.

Reply:
I limited the prefix to be 6-8 characters - perhaps this should just be 6 or more. I guess there is no reason for an upper limit. As it is, anything length > 8 that is not a valid model UUID will return a not found error if it isn't a model name. Should we loosen the check?

Some of the tests for specific commands flex using a full model UUID. But where the test set-up uses invalid UUIDs, the limit above means we can't test them if they are not 6-8 long.